src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexRustTests.cs

```diff
@@ -0,0 +1,443 @@
+                yield return new object[] { engine, @"a+b+c", "aabbabc", new[] { (4, 7) } };
+                yield return new object[] { engine, @"a*", "aaa", new[] { (0, 3), (3, 3) } };
+                yield return new object[] { engine, @"(a*)*", "-", new[] { (0, 0), (1, 1) } };
+                yield return new object[] { engine, @"(a*)+", "-", new[] { (0, 0), (1, 1) } };
+                yield return new object[] { engine, @"(a*|b)*", "-", new[] { (0, 0), (1, 1) } };
+                yield return new object[] { engine, @"(a+|b)*", "ab", new[] { (0, 2), (2, 2) } };
+                yield return new object[] { engine, @"(a+|b)+", "ab", new[] { (0, 2) } };
+                yield return new object[] { engine, @"(a+|b)?", "ab", new[] { (0, 1), (1, 2), (2, 2) } };
+                yield return new object[] { engine, @"[^ab]*", "cde", new[] { (0, 3), (3, 3) } };
+                yield return new object[] { engine, @"(^)*", "-", new[] { (0, 0), (1, 1) } };
+                yield return new object[] { engine, @"a*", "", new[] { (0, 0) } };
+                yield return new object[] { engine, @"([abc])*d", "abbbcd", new[] { (0, 6) } };
+                yield return new object[] { engine, @"([abc])*bcd", "abcd", new[] { (0, 4) } };
+                yield return new object[] { engine, @"a|b|c|d|e", "e", new[] { (0, 1) } };
+                yield return new object[] { engine, @"(a|b|c|d|e)f", "ef", new[] { (0, 2) } };
+                yield return new object[] { engine, @"((a*|b))*", "-", new[] { (0, 0), (1, 1) } };
+                yield return new object[] { engine, @"abcd*efg", "abcdefg", new[] { (0, 7) } };
+                yield return new object[] { engine, @"ab*", "xabyabbbz", new[] { (1, 3), (4, 8) } };
+                yield return new object[] { engine, @"ab*", "xayabbbz", new[] { (1, 2), (3, 7) } };
+                yield return new object[] { engine, @"(ab|cd)e", "abcde", new[] { (2, 5) } };
+                yield return new object[] { engine, @"[abhgefdc]ij", "hij", new[] { (0, 3) } };
+                yield return new object[] { engine, @"(a|b)c*d", "abcd", new[] { (1, 4) } };
+                yield return new object[] { engine, @"(ab|ab*)bc", "abc", new[] { (0, 3) } };
+                yield return new object[] { engine, @"a([bc]*)c*", "abc", new[] { (0, 3) } };
+                yield return new object[] { engine, @"a[bcd]*dcdcde", "adcdcde", new[] { (0, 7) } };
+                yield return new object[] { engine, @"(ab|a)b*c", "abc", new[] { (0, 3) } };
+                yield return new object[] { engine, @"[A-Za-z_][A-Za-z0-9_]*", "alpha", new[] { (0, 5) } };
+                yield return new object[] { engine, @"^a(bc+|b[eh])g|.h$", "abh", new[] { (1, 3) } };
+                yield return new object[] { engine, @"abcd", "abcd", new[] { (0, 4) } };
+                yield return new object[] { engine, @"a(bc)d", "abcd", new[] { (0, 4) } };
+                yield return new object[] { engine, @"a[☺-♥]?c", "a☻c", new[] { (0, 3) } };
```

Comment:
Can we use \uxxxx instead of non-ASCII characters?

Reply:
Yup, I thought I had built my reader to automatically convert those, not sure why they weren't converted here.